DQM/CSCMonitorModule/plugins/CSCDQM_StripClusterFinder.cc

```diff
@@ -189,84 +190,84 @@ namespace cscdqm {
+            (c2.IC2MAX >= c1.IC1MIN && c2.IC2MAX <= c1.IC1MAX && c2.JC2MAX >= c1.JC1MIN && c2.JC2MAX <= c1.JC1MAX)) {
+          icstart = KillCluster(ic1, ic2, c1, c2);
           return true;
         } else {
-          if ((IC1MIN >= IC2MIN && IC1MIN <= IC2MAX && JC1MIN >= JC2MIN && JC1MIN <= JC2MAX) ||
-              (IC1MIN >= IC2MIN && IC1MIN <= IC2MAX && JC1MAX >= JC2MIN && JC1MAX <= JC2MAX) ||
-              (IC1MAX >= IC2MIN && IC1MAX <= IC2MAX && JC1MIN >= JC2MIN && JC1MIN <= JC2MAX) ||
-              (IC1MAX >= IC2MIN && IC1MAX <= IC2MAX && JC1MAX >= JC2MIN && JC1MAX <= JC2MAX)) {
-            KillCluster();
+          if ((c1.IC1MIN >= c2.IC2MIN && c1.IC1MIN <= c2.IC2MAX && c1.JC1MIN >= c2.JC2MIN && c1.JC1MIN <= c2.JC2MAX) ||
+              (c1.IC1MIN >= c2.IC2MIN && c1.IC1MIN <= c2.IC2MAX && c1.JC1MAX >= c2.JC2MIN && c1.JC1MAX <= c2.JC2MAX) ||
+              (c1.IC1MAX >= c2.IC2MIN && c1.IC1MAX <= c2.IC2MAX && c1.JC1MIN >= c2.JC2MIN && c1.JC1MIN <= c2.JC2MAX) ||
+              (c1.IC1MAX >= c2.IC2MIN && c1.IC1MAX <= c2.IC2MAX && c1.JC1MAX >= c2.JC2MIN && c1.JC1MAX <= c2.JC2MAX)) {
+            icstart = KillCluster(ic1, ic2, c1, c2);
             return true;
           }
         }
       }
     }
     return false;
   }
-  void StripClusterFinder::KillCluster(void) {
+  uint32_t StripClusterFinder::KillCluster(uint32_t ic1, uint32_t ic2, C1 const& c1, C2 const& c2) {
     // Match Clusters and kill one of clusters.
-    if (IC1MIN < IC2MIN)
-      MEStripClusters[ic1].LFTBNDStrip = IC1MIN;
+    if (c1.IC1MIN < c2.IC2MIN)
+      MEStripClusters[ic1].LFTBNDStrip = c1.IC1MIN;
     else
-      MEStripClusters[ic1].LFTBNDStrip = IC2MIN;
-    if (JC1MIN < JC2MIN)
-      MEStripClusters[ic1].LFTBNDTime = JC1MIN;
+      MEStripClusters[ic1].LFTBNDStrip = c2.IC2MIN;
+    if (c1.JC1MIN < c2.JC2MIN)
+      MEStripClusters[ic1].LFTBNDTime = c1.JC1MIN;
     else
-      MEStripClusters[ic1].LFTBNDTime = JC2MIN;
-    if (IC1MAX > IC2MAX)
-      MEStripClusters[ic1].IRTBNDStrip = IC1MAX;
+      MEStripClusters[ic1].LFTBNDTime = c2.JC2MIN;
+    if (c1.IC1MAX > c2.IC2MAX)
+      MEStripClusters[ic1].IRTBNDStrip = c1.IC1MAX;
     else
-      MEStripClusters[ic1].IRTBNDStrip = IC2MAX;
-    if (JC1MAX > JC2MAX)
-      MEStripClusters[ic1].IRTBNDTime = JC1MAX;
+      MEStripClusters[ic1].IRTBNDStrip = c2.IC2MAX;
+    if (c1.JC1MAX > c2.JC2MAX)
+      MEStripClusters[ic1].IRTBNDTime = c1.JC1MAX;
     else
-      MEStripClusters[ic1].IRTBNDTime = JC2MAX;
+      MEStripClusters[ic1].IRTBNDTime = c2.JC2MAX;
 
     MEStripClusters.erase(MEStripClusters.begin() + ic2);
-    icstart = ic1;
-
-    return;
+    return ic1;
   }
   void StripClusterFinder::RefindMax(void) {
-    int iLS, iRS, iLT, iRT;
-    int iS, jT;
-    int ilocal;
-    float GlobalMax;
-    bool Erased;
+    //int iLS, iRS, iLT, iRT;
+    //int iS, jT;
+    //int ilocal;
+    //float GlobalMax;
+    //bool Erased;
```

Comment:
As this is a cleanup, the following lines could be removed
```suggestion
```

Reply:
done